Set the SolrCloud CRD we create to use ephemeral volumes

mogul opened the pull request:
*Should* address https://github.com/GSA/datagov-deploy/issues/3071.

mogul commented:
It's super-weird that this same error did not happen when running the tests with KinD locally. 🤔 
```
provision:   failed - Error: release solr-7e317e2a23a3574f failed, and has been uninstalled due to atomic being set: SolrCloud.solr.bloomberg.com "solr-7e317e2a23a3574f" is invalid: spec.dataStorage.ephemeral: Invalid value: "null": spec.dataStorage.ephemeral in body must be of type object: "null"  on main.tf line 40, in resource "helm_release" "solrcloud":  40: resource "helm_release" "solrcloud" { exit status 1
provision:   done
Failed to bind to service instance API request error 500: &{ Error: Failed to read Ingress '/' because: ingresses.extensions "solr-7e317e2a23a3574f-solrcloud-common" not found  on data.tf line 8, in data "kubernetes_ingress" "solrcloud-ingress":   8: data "kubernetes_ingress" "solrcloud-ingress" { exit status 1}
make: *** [Makefile:57: demo-up] Error 1
Error: Process completed with exit code 2.
```

adborden commented:
What's the tarball for?

adborden commented:
> What's the [tarball](https://github.com/GSA/datagov-brokerpak/pull/17/files#diff-b407e80f4fbc59a645ce3dc249a4f65e1454e9c17754feba16bafc4687518777) for?

**rubs eyes to see clearly** Ah, I thought the tarball/helm chart is new, but now I realize you're just updating it. Got thrown by the GH treatment for updating binary files.

BUT, it doesn't look like that tarball is being used at all and I don't see why we're committing it to source control. Since the [terraform](https://github.com/GSA/datagov-brokerpak/blob/main/terraform/provision/main.tf#L42) is always pulling from this [old helm-chart-release tag](https://github.com/GSA/datagov-brokerpak/releases/tag/helm-chart-release) from Oct 2020, I'm not seeing how the tarball in the repo is significant. Feels like it should be git ignored.

Is there documentation on how to build/release the helm chart? It seems like it's a manual step. I would say, let's document it, but I feel like adding the CI workflow to publish it on merges to main is just as easy.

I'm also thinking about your slack comment and if the helm chart should be its own repository and if that would make testing/publishing easier or not.

mogul commented:
Once again ready for review. A summary of changes: 
* I added our changes to explicitly set up a zookeper instance using ephemeral storage, and then have solrcloud use it
* I added out changes to make it easier to just run `terraform init; terraform apply` in `terraform/provision`, which shortens to iteration loop
* I configured the `solr-operator` to stop exposing individual SolrCloud nodes since we don't need that
* I changed the GitHub Actions so that releases are based on pushes of tags rather than merges into main, and include the Helm chart tarball
* I changed the `helm_release` to grab the _latest_ release's `solr-crd` Helm chart tarball (not crazy about this; can't think of a way to get the brokerpak's manifest id from within the Terraform so I can get the exact corresponding version)

To test the release workflow I tagged the final commit as 0.19.0 and pushed the tag. [The tag was released successfully](https://github.com/GSA/datagov-brokerpak/actions/runs/758028845), and [the resulting release includes the Helm chart tarball](https://github.com/GSA/datagov-brokerpak/releases/tag/0.19.0). 🥳 

Note: I **did** try removing the use of the Helm provider and chart altogether, instead using the `kubernetes-alpha` provider.  That works on the Terraform side, and I pushed [a branch ahead of this one that makes that change](https://github.com/GSA/datagov-brokerpak/tree/drop-use-of-helm-provider-and-chart). However, the `kubernetes-alpha` provider requires a minimum Terraform version of 0.14.8, and the CSB only works with 0.12.x! So that branch will have to wait until [the CSB works with newer Terraform versions](https://github.com/cloudfoundry-incubator/cloud-service-broker/issues/99).

mogul commented:
I asked upstream that the solr-operator configure the zookeeper-operator to use ephemeral storage when it's configured to use ephermeral storage itself.
https://github.com/apache/solr-operator/issues/259

adborden commented on .github/workflows/release.yml:
```diff
@@ -0,0 +1,38 @@
+      run: mv datagov-services-pak-current.brokerpak datagov-services-pak-${GITHUB_REF#refs/*/}.brokerpak
+
+    - name: Create helm chart tarball
+      run: cd terraform/provision && tar czvf ../../solr-crd.tar.gz solr-crd
```
We should move this to the Makefile so it's straightforward to build the release locally (and in case we want to sign it in the future). Also add the tar.gz to .gitignore.
